torch/quantization/fx/_equalize.py

```diff
@@ -347,6 +366,68 @@ def scale_weight_node(
     scaled_bias = torch.mul(bias, next_equalization_scale)
     modules[node.target].bias = nn.Parameter(scaled_bias)
 
+def scale_weight_functional(
+    op_node: Node,
+    model: GraphModule,
+    modules: Dict[str, nn.Module],
+    equalization_scale: torch.Tensor,
+) -> None:
+    """ Scales the weight value for functional layers
+    """
+
+    # From the given op_node, the path looks like:
+    #   get_attr(weight) -> weight_quant_obs -> weight_eq_obs -> op_node
+    # So we want to trace back from the op_node to get the equalization observer
+    # node, then the quantization observer node, and then finally the weight
+    # node which contains the weight values.
+
+    # Get the equalization observer node
+    weight_eq_obs_node = maybe_get_weight_eq_obs_node(op_node, modules)
+    if weight_eq_obs_node is None:
+        return
+
+    # Get the quantization observer node
+    weight_quant_obs_node = weight_eq_obs_node.args[0]
+    if weight_quant_obs_node is None:
+        return
+    assert(isinstance(weight_quant_obs_node, Node) and
+           isinstance(modules[str(weight_quant_obs_node.target)], ObserverBase))
+
+    # Get the get_attr(weight) node
+    weight_node = weight_quant_obs_node.args[0]
+    if weight_node is None:
+        return
+    assert(isinstance(weight_node, Node) and weight_node.op == 'get_attr')
+
+    parent_name, name = _parent_name(weight_node.target)
+    weight = getattr(modules[parent_name], name)
+
+    # Scale the weights for input-weight equalization
+    # If the following layer needs to be equalized then we will multiply its scale
+    scaled_weight = torch.mul(weight, torch.reciprocal(equalization_scale))
+    # TODO: connecting functional layers?
+    # scaled_weight = torch.mul(scaled_weight, next_equalization_scale)
+
+    setattr(modules[parent_name], name, scaled_weight)
+
+def clear_weight_quant_obs_node(op_node: Node, modules: Dict[str, nn.Module]) -> None:
+    """ Given the operation node, we want find the corresponding quantization
+    observer and reset its min/max values
+    """
+    weight_eq_obs_node = maybe_get_weight_eq_obs_node(op_node, modules)
+    if weight_eq_obs_node is None:
+        return
+
+    weight_quant_obs_node = weight_eq_obs_node.args[0]
+    if weight_quant_obs_node is None:
+        return
+    assert(isinstance(weight_quant_obs_node, Node))
+
+    weight_quant_obs = modules[str(weight_quant_obs_node.target)]
+    assert(isinstance(modules[str(weight_quant_obs_node.target)], ObserverBase))
+    weight_quant_obs.min_val = torch.tensor(float("inf"))
```

Comment:
I feel it's better to add the clear method in weight observer instead, resetting to "inf" is an implementation detail for the weight observer, it's better not to leak it here.

Reply:
I have addressed this comment in this PR: https://github.com/pytorch/pytorch/pull/60883